superset/viz.py

```diff
@@ -106,10 +106,11 @@ def get_df(self, query_obj=None):
                 timestamp_format = dttm_col.python_date_format
 
         # The datasource here can be different backend but the interface is common
-        self.results = self.datasource.query(**query_obj)
+        query_str, engine, start_dttm = self.datasource.get_query_str(**query_obj)
+        self.query = query_str
+        self.results = self.datasource.query(query_str, engine, start_dttm)
```

Comment:
The engine can probably be fetched within the scope of that function, and start_dttm can probably generated there also.